### DISCUSSION TITLE
Time_limit is not working

### DISCUSSION THREAD

**Discussion by LeoAlle** [google/or-tools#3568] [Routing (and legacy CP) questions]
**What version of OR-Tools and what language are you using?**
Version: v9.4
Language: Python

**Which solver are you using (e.g. CP-SAT, Routing Solver, GLOP, BOP, Gurobi)**
ROUTING SOLVER (routing_enums_pb2)

**What operating system (Linux, Windows, ...) and version?**
Windows 10 Home 21H2

**What did you do?**
I've put a time_limit of 120 seconds with this line of code:
```
search_parameters.time_limit.seconds = 120
```
When I run, it doesn't stop after that time limit and then, after a while, it crashes.

**Comment by Mizux** [comment]
If you enable the log search `search_parameters.log_search = True`
Did you see some solutions trace ?

usually this limit is needed/used to stop the guided local search but in your case maybe the solver can't find a first solution and continue to search until it is out of memory...

**Reply by LeoAlle** [reply]
Yes I've also tried to enable log search, but the situation remains the same.
And I've also tried to put guided local search as searching solution strategy.

**Reply by Regista6** [reply]
What's the value of  ```routing.status()```?. Probably that can give you some idea.

**Reply by LeoAlle** [reply]
It seems to stop in the line where it searches for a solution.
solution = routing.SolveWithParameters(search_parameters)
If I print the routing status just before that line it prints "0".

**Comment by Regista6** [comment]
Maybe try changing the first solution strategies or adding disjunction constraints to make nodes droppable to get at least a partial initial solution that you can feed to GLS to get a complete solution.

**Reply by LeoAlle** [reply]
I'm working on it and I will let you know. Up to now, it still doesn't work.

**Reply by LeoAlle** [reply]
I've checked the input's types and they are correct, the code is the same of the online examples. I've tried to change the first solution strategy, but the kernel dies every time I try to run it. Any suggestions?
